### ISSUE TITLE
Excluded parameters produce incorrect redirect 

### ISSUE THREAD

**Issue by thampe** [nexcess/magento-turpentine#935]
Hello,

we ignore the Google utm_* parameters in our Turpentine configuration, which looks like this in our deployed vcl:
```
if (true && req.url ~ "[?&](utm_source|utm_medium|utm_campaign|utm_content|utm_term|gclid|cx|ie|cof|siteurl)=") {
set req.url = regsuball(req.url, "(?:(\?)?|&)(?:utm_source|utm_medium|utm_campaign|utm_content|utm_term|gclid|cx|ie|cof|siteurl)=[^&]+", "\1");
set req.url = regsuball(req.url, "(?:(\?)&|\?$)", "\1");
}
```

This is required to ensure that users from Google campaigns also get a fast cached page, instead of a cache miss.
But in some cases we - as many other online shops - need to do some redirects. In our use case we have to redirect the URL http://www.example.com/de/bar.html?utm_source=123 to http://www.example.com/bar.html?utm_source=123 on the backend server (Apache mod_rewrite rule), which would work, if the varnish configuration would not strip the ignored (for caching) utm parameters from the request URI, since varnish/turpentine does not cache 301 redirects at all.
So on the backend server the request URI is just: /de/bar.html

So Turpentine needs an option to also pass ignored parameters to the backend server, if the status code was 301.

**Comment by miguelbalparda** [comment]
How are you configuring the redirects? are you adding to the original url the UTM params?

**Comment by thampe** [comment]
Exactly, we are adding the UTM parameters back. Our .htaccess rule looks like this:

```
RewriteCond %{HTTP_HOST} example\.com$
RewriteCond %{REQUEST_URI} ^\/de
RewriteRule ^de\/(.*)$ http://www.example.com/$1 [R=301,L]
```

**Comment by thampe** [comment]
Fixed in #945
